tests/5.0/requires/test_requires_unified_shared_memory_heap_is_device_ptr.c

```diff
@@ -0,0 +1,71 @@
+//===---test_requires_unified_shared_memory.c -------------------------------===//
+//
+// OpenMP API Version 5.0 Nov 2018
+// 
+// This test Checks for unified shared memory of an array that is allocated on 
+// the heap and that is accessed from host and device with the same pointer
+//
+// To guarantee the use of the device pointer, we use the is_device_ptr clause
+//
+////===----------------------------------------------------------------------===//
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1024
+
+#pragma omp requires unified_shared_memory
+
+int unified_shared_memory_heap() {
+  OMPVV_INFOMSG("Unified shared memory testing - Array on heap");
+  int errors = 0;
+  
+  int *anArray;
+  int anArrayCopy[N];
+
+  anArray = (int*)malloc(sizeof(int)*N);
```

Comment:
Don't we need to use omp device allocation ?

Reply:
So the answer is yes, if the unified memory requirement is satisfied and working correctly the heap allocation should be accessible as well as "see" the same value.